src/controller.ts

```diff
@@ -0,0 +1,133 @@
+//backend application framework for nodejs
+import {Express} from 'express';
+import * as userdata from './data';
+import { isSimpleUser, isIdentifiedUser } from './user';
+
+/**
+ * Broadcast-function 
+ * 
+ * @callback callbackBroadcast
+ * @param {string} tobeBroadcast The string, that is to be broadcast
+ */
+
+export class UserController {
+    private users: userdata.UserRepository;
+
+    /** 
+     * Creates an instance of UserController
+     * 
+     * @param {express.Express} app The express-app whose requests should be handled
+     * @param {callbackBroadcast} broadcast A function by which the controller can initiate a server broadcast
+     */
+    public constructor(private readonly app: Express, private broadcast: { (dataToBroadcast: string): void }) {
+        //create a UserData-Object
+        this.users = new userdata.UserRepository();
```

Comment:
It is better to name a variable like its type or its designated use. In your is something like `userRepo` a good choice.